### PR TITLE
Fix application crash when disabling cache auto-clear option and potential attack vector

### PR DESCRIPTION
**Pull request by agrahn**
When disabling the passphrase cache auto-clear setting, the cache needs to be cleared once in order to prevent a malicious user (someone knowing the screen-lock pin but not knowing the correct passphrase) from bypassing the auto-clearing after waking up/unlocking the device, followed by disabling the auto-clearing option in the settings.

However, clearing `EncryptedSharedPreferences` requires authentication, otherwise the app crashes. Currently, a crash can be provoked as follows:

1. Open APS, make sure that passphrase caching and auto-clear option are both enabled.
2. Decrypt some entry in the store with the correct passphrase.
3. Switch off the screen.
4. Unlock the device again and keep it awake for one minute or two by doing something within another app, e.g. browsing the internet.       
5. Switch to APS, go directly to the PGP settings screen and disable the auto-clear option.
6. APS crashes and immediately reloads automatically.

APS crashes because the user authentication has expired in the meantime and clearing the passphrase from `EncryptedSharedPreferences` is attempted.

After the crash, the auto-clear option is disabled but the passphrase is still present in the cache. From now, the bad user is given access to any store entry by just entering the correct screen-lock pin, without the need to also enter the PGP passphrase. In this way, that user successfully bypassed and disabled the cache auto-clearing.

This PR 
1. restores `BiometricAuthenticator` before clearing the passphrase from `EncryptedSharedPreferences`. This prevents app crashing  
2. enforces the auto-clearing option to stay enabled in the case of failed authentication (caused by dismissing the authentication dialog). This prevents the attempt of bypassing cache auto-clearing.

### CODE REVIEW

File: app/src/main/java/app/passwordstore/ui/settings/PGPSettings.kt [hunk 1]
```diff
@@ -75,11 +75,35 @@ class PGPSettings(
...
+                  activity.sharedPrefs.edit {
+                    activity.sharedPrefs.edit { remove(PreferenceKeys.CLEAR_PASSPHRASE_CACHE) }
+                  }
+                }
+              }
+            } else {
+              activity.sharedPrefs.edit {
+                activity.sharedPrefs.edit { remove(PreferenceKeys.CLEAR_PASSPHRASE_CACHE) }
+              }
+            }
```
Comment (msfjarvis) [inline app/src/main/java/app/passwordstore/ui/settings/PGPSettings.kt]: Nesting two calls to `edit` is redundant.

You might need to reformat the code after applying this.

```suggestion
                  activity.sharedPrefs.edit {
                    remove(PreferenceKeys.CLEAR_PASSPHRASE_CACHE)
                  }
                }
              }
            } else {
              activity.sharedPrefs.edit {
                remove(PreferenceKeys.CLEAR_PASSPHRASE_CACHE)
              }
            }
```

**Review by msfjarvis — APPROVED** [review]

File: app/src/main/java/app/passwordstore/ui/settings/PGPSettings.kt (hunk 1 above)
Comment (agrahn) [reply app/src/main/java/app/passwordstore/ui/settings/PGPSettings.kt:105]: Hm, strange. In my latest local version that I force-pushed, this nesting is not present, but on GitHub it indeed looks as you noticed. Maybe I should delete the branch on GitHub and push a corrected version.

Comment (agrahn) [reply app/src/main/java/app/passwordstore/ui/settings/PGPSettings.kt]: Will fix it in a moment.

**Review by msfjarvis — APPROVED** [review]

File: app/src/main/java/app/passwordstore/ui/settings/PGPSettings.kt (hunk 1 above)
Comment (agrahn) [reply app/src/main/java/app/passwordstore/ui/settings/PGPSettings.kt]: Should be ok now.

**Pull request merged**